frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view_helper.tsx

```diff
@@ -35,6 +39,52 @@ export function getBaseSegmentationName(segmentationLayer: APIDataLayer) {
   );
 }
 
+export function getVolumeRequestUrl(
+  dataset: APIDataset,
+  tracing: HybridTracing | null,
+  tracingId: string | undefined,
+  visibleSegmentationLayer: APISegmentationLayer | APIDataLayer,
+) {
+  const datasetName = dataset.name;
+  const organization = dataset.owningOrganization;
+  const dataStoreHost = dataset.dataStore.url;
+
+  const dataStoreUrl = `${dataStoreHost}/data/datasets/${organization}/${datasetName}/layers/${visibleSegmentationLayer.name}`;
+  if (tracing == null) return dataStoreUrl;
+  const tracingStoreHost = tracing?.tracingStore.url;
+
+  const maybeVolumeTracing =
+    "tracingId" in visibleSegmentationLayer && visibleSegmentationLayer.tracingId != null
+      ? getVolumeTracingById(tracing, visibleSegmentationLayer.tracingId)
+      : null;
+  // For non-segmentation layers and for viewing datasets, we'll always use the datastore URL
+  const shouldUseDataStore = maybeVolumeTracing == null;
+  return shouldUseDataStore ? dataStoreUrl : `${tracingStoreHost}/tracings/volume/${tracingId}`;
+}
```

Comment:
I think, this function could be simpler. How about this?

```suggestion
export function getVolumeRequestUrl(
  dataset: APIDataset,
  tracing: HybridTracing | null,
  tracingId: string | undefined,
  visibleSegmentationLayer: APISegmentationLayer | APIDataLayer,
) {
  if (tracing == null || tracingId == null) {
    return `${dataset.dataStore.url}/data/datasets/${dataset.owningOrganization}/${dataset.name}/layers/${visibleSegmentationLayer.name}`;
  } else {
    const tracingStoreHost = tracing.tracingStore.url;
    return `${tracingStoreHost}/tracings/volume/${tracingId}`;
  }
}
```

Your current implementation is quite pessismistic and doesn't trust the provided tracingId. However, I'd argue that that ID can be trusted (it's typically passed as `volumeTracing?.tracingId` anyway). If the caller provides a wrong id, that's a bug that should be fixed then.